core/src/main/kotlin/fr/sncf/osrd/standalone_sim/etcsContextBuilder.kt

```diff
@@ -10,22 +12,42 @@ import fr.sncf.osrd.utils.distanceRangeSetOf
 import fr.sncf.osrd.utils.indexing.DirStaticIdx
 import fr.sncf.osrd.utils.indexing.StaticIdxList
 import fr.sncf.osrd.utils.units.Offset
+import fr.sncf.osrd.utils.units.meters
 
 /** Build the ETCS context, if relevant. */
 fun makeETCSContext(
     rollingStock: RollingStock,
-    infra: RawInfra,
+    infra: FullInfra,
     chunkPath: ChunkPath,
-    routePath: StaticIdxList<Route>
+    routePath: StaticIdxList<Route>,
+    blockPath: StaticIdxList<Block>
 ): EnvelopeSimContext.ETCSContext? {
-    val isRollingStockETCSCompatible = false // TODO
-    if (!isRollingStockETCSCompatible) return null
+    val blockInfra = infra.blockInfra
+    val etcsRanges = distanceRangeSetOf()
+    val etcsLevel2 =
+        infra.signalingSimulator.sigModuleManager.findSignalingSystemOrThrow(ETCS_LEVEL2.id)
+    var offsetSinceStart = Offset<Path>(0.meters)
```

Comment:
I'm afraid it's not as simple as that, the path may start in the middle of a block. We'd need to identify where the path starts exactly, based on `chunkPath.beginOffset`. That includes matching the blocks with the chunks, it's a little tedious.